.semgrep.yml

```diff
@@ -235,6 +235,60 @@ rules:
         - pattern: if $VALUE, $OK := d.GetOk($KEY); $OK && len($VALUE.(string)) > 0 { $BODY }
     severity: WARNING
 
+  - id: helper-schema-ResourceData-SetId-empty-without-IsNewResource-check
+    languages: [go]
+    message: Calling `d.SetId("")` should ensure `!d.IsNewResource()` is checked first
+    paths:
+      exclude:
+        - aws/resource_aws_api_gateway_*.go
+        - aws/resource_aws_apigatewayv2_*.go
+        - aws/resource_aws_app_cookie_stickiness_policy.go
+        - aws/resource_aws_appautoscaling_*.go
+        - aws/resource_aws_appmesh_*.go
+        - aws/resource_aws_appsync_*.go
+        - aws/resource_aws_athena_*.go
+        - aws/resource_aws_autoscaling_*.go
+        - aws/resource_aws_autoscalingplans_scaling_plan.go
+        - aws/resource_aws_b*.go
+        - aws/resource_aws_c*.go
+        - aws/resource_aws_d*.go
+        - aws/resource_aws_e*.go
+        - aws/resource_aws_f*.go
+        - aws/resource_aws_g*.go
```

Comment:
are these explicit lines in place of usage like: `aws/resource_aws_[b-g]*.go`?

Reply:
i tried using the range usage locally and the report didn't send back any errors from those files 🤔 so maybe it works?

Reply:
It does seem to work! 👍  Will adjust by rebasing this with that change.